openedx/core/djangoapps/course_groups/cohorts.py

```diff
@@ -235,47 +211,76 @@ def get_cohort(user, course_key, assign=True):
         if not assign:
             return None
 
-    choices = course.auto_cohort_groups
-    if len(choices) > 0:
-        # Randomly choose one of the auto_cohort_groups, creating it if needed.
-        group_name = local_random().choice(choices)
+    course_cohorts = get_course_cohorts(course, assignment_type=CourseUserGroup.RANDOM)
+    if course_cohorts:
+        course_cohort = local_random().choice(course_cohorts)
     else:
-        # Use the "default cohort".
-        group_name = DEFAULT_COHORT_NAME
+        course_cohort, __ = CourseUserGroup.objects.get_or_create(
+            course_id=course_key,
+            group_type=CourseUserGroup.COHORT,
+            name=DEFAULT_COHORT_NAME,
+            defaults={'assignment_type': CourseUserGroup.RANDOM}
+        )
 
-    group, __ = CourseUserGroup.objects.get_or_create(
-        course_id=course_key,
-        group_type=CourseUserGroup.COHORT,
-        name=group_name
+    user.course_groups.add(course_cohort)
+
+    return course_cohort
+
+
+def migrate_cohort_settings(course):
```

Comment:
We DO currently export cohort settings now though (with the Advanced Setting in Studio)-- if cohorting is enabled, and the names of random cohorts). This change in behavior needs further discussion, @explorerleslie.

Not only will the changes that have been made in the instructor dashboard not be exported, but if this course was an older course and had a cohort configuration defined in Advanced Settings, that value will be exported unless we explicitly specify that cohort_configuration should not be exported. I assume it is not possible to completely remove the field from the code, as that would break older courses, so I was imagining that you would simply add the cohort_configuration setting to the list of things that are not show in Advanced Settings.